File: transcrypt/development/shipment/docs/code/classes.js
var A = __class__ ('A', [object], {
	get __init__ () {return __get__ (this, function (self, x) {
		self.x = x;
	});},
	get show () {return __get__ (this, function (self, label) {
		print ('A.show', label, self.x);
	});}
});
var B = __class__ ('B', [object], {
	get __init__ () {return __get__ (this, function (self, y) {
		alert ('In B constructor');
		self.y = y;
	});},
	get show () {return __get__ (this, function (self, label) {
		print ('B.show', label, self.y);
	});}
});
var C = __class__ ('C', [A, B], {
	get __init__ () {return __get__ (this, function (self, x, y) {
		alert ('In C constructor');
		A.__init__ (self, x);
		B.__init__ (self, y);
		self.show ('constructor');
	});},
	get show () {return __get__ (this, function (self, label) {
		B.show (self, label);
		print ('C.show', label, self.x, self.y);
	});}
});
var a = A (1001);
a.show ('america');
var b = B (2002);
b.show ('russia');
var c = C (3003, 4004);
c.show ('netherlands');
var show2 = c.show;
show2 ('copy');

File: Transcrypt-3.5.196/transcrypt/docs/code/classes.py
class A:
	def __init__ (self, x):
		self.x = x

	def show (self, label):
		print ('A.show', label, self.x)
	
class B:
	def __init__ (self, y):
		alert ('In B constructor')
		self.y = y
		
	def show (self, label):
		print ('B.show', label, self.y)
		
class C (A, B):
	def __init__ (self, x, y):
		alert ('In C constructor')
		A.__init__ (self, x)
		B.__init__ (self, y)
		self.show ('constructor')
		
	def show (self, label):
		B.show (self, label)
		print ('C.show', label, self.x, self.y)
	
a = A (1001)
a.show ('america')

b = B (2002)
b.show ('russia')

c = C (3003, 4004)
c.show ('netherlands')

show2 = c.show
show2 ('copy')
